runtime/src/types.rs

```diff
@@ -228,7 +230,28 @@ pub struct CheckTxResult {
     pub error: Error,
 
     #[serde(rename = "meta")]
-    pub meta: Option<cbor::Value>,
+    pub meta: Option<CheckTxMetadata>,
+}
+
+/// CheckTx transaction metadata.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+pub struct CheckTxMetadata {
+    #[serde(default)]
```

Comment:
Probably should add the `#[serde(skip_serializing_if = "num_traits::Zero::is_zero")]` to be consistent.